pkg/chain/ethereum/ethutil/signer.go

```diff
@@ -0,0 +1,146 @@
+package ethutil
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// SignatureSize is a byte size of a signature calculated by Ethereum with
+// recovery-id, V, included. The signature consist of three values (R,V,S)
+// in the following order:
+// R = [0:31]
+// S = [32:63]
+// V = [64]
+const SignatureSize = 65
+
+// EthereumSigner provides functions to sign a message and verify a signature
+// using the Ethereum-specific signature format. It also provides functions for
+// convertion of a public key to address.
+type EthereumSigner struct {
+	operatorKey *ecdsa.PrivateKey
+}
+
+// NewSigner creates a new EthereumSigner instance for the provided private key.
+func NewSigner(privateKey *ecdsa.PrivateKey) *EthereumSigner {
+	return &EthereumSigner{privateKey}
+}
+
+// PublicKey returns byte representation of a public key for the private key
```

Comment:
// PublicKey returns byte representation of a public key which is part of the signer's private key.

^ up to you

Reply:
I'd prefer to leave it as-is, without going too deep into how ECDSA private/public key works.